plugins/modules/supervisorctl.py

```diff
@@ -199,22 +209,23 @@ def get_matched_processes():
             matched.append((process_name, status))
         return matched
 
-    def take_action_on_processes(processes, status_filter, action, expected_result):
+    def take_action_on_processes(processes, status_filter, action, expected_result, exit_module=True):
         to_take_action_on = []
         for process_name, status in processes:
             if status_filter(status):
                 to_take_action_on.append(process_name)
 
-        if len(to_take_action_on) == 0:
+        if len(to_take_action_on) == 0 and exit_module:
             module.exit_json(changed=False, name=name, state=state)
```

Comment:
This should probably rather be:
```suggestion
        if len(to_take_action_on) == 0:
            if not exit_module:
                return
            module.exit_json(changed=False, name=name, state=state)
```
Otherwise the code below is run if `exit_module` is `False` and `to_take_action_on` is empty.

Similar below in the next line.